GenderPayGap.WebUI/Views/GovUkFrontend/GovUkFrontendLayout.cshtml

```diff
@@ -55,81 +55,88 @@
 
     @{
         List<HeaderNavigationViewModel> headerNavigationLinks;
-        if (FeatureFlagHelper.IsFeatureEnabled(FeatureFlag.ReportingStepByStep))
+        if (ViewBag.ServiceUnavailable == true)
         {
-            headerNavigationLinks = new List<HeaderNavigationViewModel>
-            {
-                new HeaderNavigationViewModel
-                {
-                    Text = "Search and compare",
-                    Href = Url.Action("SearchResults", "Viewing")
-                },
-                new HeaderNavigationViewModel
-                {
-                    Text = "Download",
-                    Href = Url.Action("Download", "Viewing")
-                },
-                new HeaderNavigationViewModel
-                {
-                    Text = "Close the gap",
-                    Href = Url.Action("Overview", "ActionHub")
-                },
-                new HeaderNavigationViewModel
-                {
-                    Text = "Employer guidance",
-                    Href = Url.Action("StepByStepStandalone", "ReportingStepByStep")
-                }
-            };
+            headerNavigationLinks = new List<HeaderNavigationViewModel>();
         }
         else
         {
-             headerNavigationLinks = new List<HeaderNavigationViewModel>
+            if (FeatureFlagHelper.IsFeatureEnabled(FeatureFlag.ReportingStepByStep))
             {
-                new HeaderNavigationViewModel
+                headerNavigationLinks = new List<HeaderNavigationViewModel>
                 {
-                    Text = "Search and compare",
-                    Href = Url.Action("SearchResults", "Viewing")
-                },
-                new HeaderNavigationViewModel
-                {
-                    Text = "Download",
-                    Href = Url.Action("Download", "Viewing")
-                },
-                new HeaderNavigationViewModel
+                    new HeaderNavigationViewModel
+                    {
+                        Text = "Search and compare",
+                        Href = Url.Action("SearchResults", "Viewing")
+                    },
+                    new HeaderNavigationViewModel
+                    {
+                        Text = "Download",
+                        Href = Url.Action("Download", "Viewing")
+                    },
+                    new HeaderNavigationViewModel
+                    {
+                        Text = "Close the gap",
+                        Href = Url.Action("Overview", "ActionHub")
+                    },
+                    new HeaderNavigationViewModel
+                    {
+                        Text = "Employer guidance",
+                        Href = Url.Action("StepByStepStandalone", "ReportingStepByStep")
+                    }
+                };
+            }
+            else

```

Comment:
Does it worth commonising the code in the if (FeatureFlagHelper.IsFeatureEnabled(FeatureFlag.ReportingStepByStep)) and its else using list.add?

Reply:
Good idea - I've made this change